### PR TITLE
Add count distinct on the Connection aggregation filter.

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description

Following the bug report https://github.com/neo4j/graphql/issues/6005, this PR adds tests and applies distinct on the count for nodes when defined explicitly with the count nodes syntax inside the Connection input object. The legacy version of aggregations continues to operate as before.

### CODE REVIEW

File: packages/graphql/src/translate/queryAST/factory/FilterFactory.ts [hunk 1]
```diff
@@ -809,8 +810,20 @@ export class FilterFactory {
         return this.wrapMultipleFiltersInLogical(filterTruthy(filterASTs));
     }
 
-    private createCountFilter(operatorKey: string, value: unknown, attachedTo: "node" | "relationship"): CountFilter {
+    private createCountFilter(
+        operatorKey: string,
+        value: unknown,
+        attachedTo: "node" | "relationship",
+        useDeprecated = true
+    ): CountFilter {
```
Comment (angrykoala) [inline packages/graphql/src/translate/queryAST/factory/FilterFactory.ts]: With 4 parameters, considering changing this to an object input to give more context when calling it

File: packages/graphql/tests/integration/deprecations/generic-filtering/types/types-localdatetime-filter-deprecated.int.test.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ describe("LocalDateTime - deprecated filters", () => {
         "should filter based on localDT comparison, for filter %s",
         async (filter) => {
             const futureId = generate({ readable: false });
-            const future = "2025-02-18T18:10:55.462Z".split("Z")[0];
+            const future = "6025-02-18T18:10:55.462Z".split("Z")[0];
```
Comment (angrykoala) [inline packages/graphql/tests/integration/deprecations/generic-filtering/types/types-localdatetime-filter-deprecated.int.test.ts:87]: ... well, this was hilarious

Setting a strict date for future is a bit of a bad practice, but I guess we don't really need to worry too much about y6k here

File: packages/graphql/tests/integration/issues/6005-filters.int.test.ts [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+            CREATE (m)<-[:ACTED_IN { screentime: 120, character: "Another Character" }]-(:${Actor} {name: "Another actor", age: 37, born: datetime('2000-02-02')})
+            CREATE (m)<-[:ACTED_IN { screentime: 10, character: "Future Terminator" }]-(arnold)
+        `);
+    });
+
+    afterEach(async () => {
+        await testHelper.close();
+    });
+
+    test("should filter movies by actors count with duplicate results (deprecated syntax)", async () => {
```
Comment (angrykoala) [inline packages/graphql/tests/integration/issues/6005-filters.int.test.ts]: We should move the deprecated syntax to the `deprecated` folder in a separate test file `6005.int.test`

File: packages/graphql/tests/integration/issues/6005-filters.int.test.ts [hunk 4]
```diff
@@ -0,0 +1,281 @@
+/*
```
Comment (angrykoala) [inline packages/graphql/tests/integration/issues/6005-filters.int.test.ts]: Per convention, this test should simply be called `6005.int.test.ts` and the name should only contain the url to the issue with the full context

File: packages/graphql/tests/integration/issues/6005-filters.int.test.ts [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+ * limitations under the License.
+ */
+
+import type { UniqueType } from "../../utils/graphql-types";
+import { TestHelper } from "../../utils/tests-helper";
+
+/**
+ * For the following tests we assuming that the deprecated syntax keep the existing behavior while when using the new syntax the
+ * distinct is applied. This is applied only to count aggregation as for node aggregations second thoughts are needed.
+ **/
```
Comment (angrykoala) [inline packages/graphql/tests/integration/issues/6005-filters.int.test.ts]: This context should be better placed in each test case, as part of the title.

File: packages/graphql/tests/tck/issues/6005-filters.test.ts [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Neo4jGraphQL } from "../../../src";
+import { formatCypher, formatParams, translateQuery } from "../utils/tck-test-utils";
+
+describe("https://github.com/neo4j/graphql/issues/6005 filters (count only)", () => {
```
Comment (angrykoala) [inline packages/graphql/tests/tck/issues/6005-filters.test.ts]: Same as the other, deprecated and new syntax should be split in 2 tests, one in deprecated folder and one here. 

The name of the file should be only the issue number, and the description the url per convention

**Review by angrykoala — CHANGES_REQUESTED** [review]

File: packages/graphql/tests/integration/issues/6005-filters.int.test.ts (hunk 3 above)
Comment (MacondoExpress) [reply packages/graphql/tests/integration/issues/6005-filters.int.test.ts]: In this case, I put it in the same file intentionally to highlight the differences between the two, not only in the API but also in the behavior. I got the feeling that by moving in two separate files, a developer without the correct context could identify one of the behaviors as a regression or a bug.

File: packages/graphql/tests/integration/deprecations/generic-filtering/types/types-localdatetime-filter-deprecated.int.test.ts (hunk 2 above)
Comment (MacondoExpress) [reply packages/graphql/tests/integration/deprecations/generic-filtering/types/types-localdatetime-filter-deprecated.int.test.ts:87]: Absolutely, this is a cherry-picked commit from the CI fix that was failing, I didn't refactor the test tho.

File: packages/graphql/tests/integration/issues/6005-filters.int.test.ts (hunk 4 above)
Comment (MacondoExpress) [reply packages/graphql/tests/integration/issues/6005-filters.int.test.ts]: This is because this tests is related only to filters, the test for 6005 should go on for the 6.x branch with the count fix projection

Comment (angrykoala) [reply packages/graphql/tests/integration/issues/6005-filters.int.test.ts]: Still, that test should eventually be merged into the same file no?

Maybe a TODO or a comment is better suited than having a renamed file?

File: packages/graphql/tests/integration/issues/6005-filters.int.test.ts (hunk 3 above)
Comment (angrykoala) [reply packages/graphql/tests/integration/issues/6005-filters.int.test.ts]: I think that the same issue number being linked with the file in `deprecated` is good context on that.

In any case, deprecated behaviours are that, deprecated, regressions on that are tested in the deprecated folder. The new one is already in the correct place.

I'm just thinking that now that we finally start to have some conventions in our tests, it is not worth breaking those just for a quite edge case confusion that may arise with some deprecated behaviour. Breaking convention will generate confusion
